closed/autoconf/custom-hook.m4

```diff
@@ -358,23 +358,24 @@ AC_DEFUN([OPENJ9_PLATFORM_SETUP],
   # Default OPENJ9_BUILD_OS=OPENJDK_BUILD_OS, but override with OpenJ9 equivalent as appropriate
   OPENJ9_BUILD_OS="${OPENJDK_BUILD_OS}"
 
-  OMR_MIXED_REFERENCES_MODE=off
-  if test "x$with_mixedrefs" != x -a "x$with_mixedrefs" != xno; then
-    if test "x$with_mixedrefs" = xyes -o "x$with_mixedrefs" = xstatic; then
-      OMR_MIXED_REFERENCES_MODE=static
-    elif test "x$with_mixedrefs" = xdynamic; then
-      OMR_MIXED_REFERENCES_MODE=dynamic
+  if test "x$with_noncompressedrefs" = xyes -o "x$with_mixedrefs" = xno; -o then
+    OMR_MIXED_REFERENCES_MODE=off
+    if test "x$with_noncompressedrefs" = xyes ; then
+      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_CPU}"
+      OPENJ9_LIBS_SUBDIR=default
     else
+      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_CPU}_cmprssptrs"
+      OPENJ9_LIBS_SUBDIR=compressedrefs
+    fi
+  else
+    OMR_MIXED_REFERENCES_MODE=static
+    if test "x$with_mixedrefs" = xdynamic; then
+      OMR_MIXED_REFERENCES_MODE=dynamic
+    elif test "x$with_mixedrefs" != xyes -o "x$with_mixedrefs" != xstatic; then
```

Comment:
* the extra `-o` on line 361 is a problem
* nit: please use a space on both sides if `;` in `if` statements

I think it is easier to follow with fewer negative tests:
```
  if test "x$with_noncompressedrefs" = xyes -o "x$with_mixedrefs" = xno ; then
    OMR_MIXED_REFERENCES_MODE=off
    if test "x$with_noncompressedrefs" = xyes ; then
      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_CPU}"
      OPENJ9_LIBS_SUBDIR=default
    else
      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_CPU}_cmprssptrs"
      OPENJ9_LIBS_SUBDIR=compressedrefs
    fi
  else
    if test "x$with_mixedrefs" = xstatic - o "x$with_mixedrefs" = xyes ; then
      OMR_MIXED_REFERENCES_MODE=static
    elif test "x$with_mixedrefs" = xdynamic ; then
      OMR_MIXED_REFERENCES_MODE=dynamic
    else
      AC_MSG_ERROR([OpenJ9 supports --with-mixedrefs=static and --with-mixedrefs=dynamic])
    fi
    OPENJ9_BUILD_MODE_ARCH="${OPENJ9_CPU}_mxdptrs"
    OPENJ9_LIBS_SUBDIR=default
  fi
```

Reply:
Makes sense. The with_mixedrefs part is basically the same as the original, not sure why I felt the need to change it.

Reply:
Ah, because it doesn't work. Another update coming.

Reply:
I've pushed a change that makes it correct although it's a bit funny. I'll let you comment on it.

Reply:
Actually it still doesn't work. It needs to handle the case where --with_mixedrefs isn't set.

Reply:
Hopefully the latest update is good.